lib/Cmts.ml

```diff
@@ -330,14 +333,31 @@ let relocate (t : t) ~src ~before ~after =
         let s = Set.add s after in
         Set.add s before )
 
+let relocate_all_to_before (t : t) ~src ~before =
```

Comment:
I really don't like duplicating code like this. Yet I see that trying to combine with `relocate_all_to_after` would be painful. Maybe just add a comment to all three `relocate` functions that the three are similar and that any bugfixes in one should consider being propagated to the others?